libmscore/tuplet.cpp

```diff
@@ -1007,7 +1006,7 @@ void Tuplet::sanitizeTuplet()
             }
       testDuration = testDuration / ratio();
       testDuration.reduce();
-      if (((testDuration - tupletDuration).reduced().numerator() != 0) || ((testDuration - baseLenDuration).reduced().numerator() != 0)) {
+      if ((testDuration - baseLenDuration).reduced().numerator() != 0) {
```

Comment:
It is still necessary to check if duration() is correct. Otherwise, tuplets with an incorrect duration() will not get sanitized if testDuration is equal to baseLenDuration. This happens in mtest/libmscore/compat206/tuplets.mscx with the nested 6:4 tuplet in measure 4. And there is a perfectly good operator for comparing fractions for equality. How about we replace this line with
```
      if (!(testDuration == baseLenDuration && baseLenDuration == duration())) {
```

Reply:
See https://github.com/musescore/MuseScore/pull/3571